src/Microsoft.Framework.Caching.SqlServer/SqlServerCacheExtensions.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Caching.SqlServer;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.DependencyInjection
+{
+    public static class SqlServerCacheExtensions
+    {
+        public static IServiceCollection AddSqlServerCache([NotNull] this IServiceCollection services)
+        {
+            services.AddOptions();
+            services.AddSingleton<IDistributedCache, SqlServerCache>();
+            return services;
+        }
+
+        public static void ConfigureSqlServerCache(
```

Comment:
Putting on my @lodejard hat, its because configuring services is seperate from configuring your app.  Its a common pattern to setup your DI in one method (ConfigureServices), and then setup the app in (Configure).  Developers are used to this paradigm.  That said, I think with the new hosting builder changes(IStartupFilter and AutoRequestServices), we have the infrastructure to add functionality which automatically registered a frameworks services and associated middleware in some ordered fashion if we wantd.  Some kind of UseFrameworks() list which contained an ordered list of frameworks who's services and middlewares were added first.